File: Syncfusion_Sample/Windows/Calculate.Windows/Samples/2.0/Working With CalcQuick Demo/CS/SelectForm.cs
#region Copyright Syncfusion Inc. 2001 - 2006
//
//  Copyright Syncfusion Inc. 2001 - 2006. All rights reserved.
//
//  Use of this code is subject to the terms of our license.
//  A copy of the current license can be obtained at any time by e-mailing
//  licensing@syncfusion.com. Any infringement will be prosecuted under
//  applicable laws. 
//
#endregion

using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace FirstSample
{
	/// <summary>
	/// Summary description for SelectForm.
	/// </summary>
	public class SelectForm : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Button manualButton;
		private System.Windows.Forms.Button autoButton;
		private System.Windows.Forms.Button button1;
		private System.Windows.Forms.Button angleFormButton;
		private System.Windows.Forms.Button button2;
		private System.Windows.Forms.Button button3;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public SelectForm()
		{
			//
			// Required for Windows Form Designer support.
			//
			InitializeComponent();

			try
			{
				System.Drawing.Icon ico = new System.Drawing.Icon(FindFile("Icon", "sfgrid.ico"));
				this.Icon = ico ;
			}
			catch
			{}


			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

        #region FindFile Utility
        /// <summary>
        /// Finds a file of the given name in the current directory or sibling "Data" directory.
        /// If file is not found, the parent folder is checked until the file is found. This method
        /// is used by our samples when they load data from a separate "Data" folder.
        /// </summary>
        /// <param name="dataDirName">The name of the "Data" folder.</param>
        /// <param name="fileName">The filename to be searched.</param>
        /// <returns>The full path of the file that was found; an empty string is returned if file is not found.</returns>

        public static string FindFile(string dataDirName, string fileName)
        {
            dataDirName = dataDirName.TrimEnd('\\', '/');
            // Check both in parent folder and Parent\Data folders.
            string dataFileName = dataDirName + "\\" + fileName;
            for (int n = 0; n < 10; n++)
            {
                if (System.IO.File.Exists(fileName))
                {
                    return fileName;
                }
                if (System.IO.File.Exists(dataFileName))
                {
                    return dataFileName;
                }
                fileName = @"..\" + fileName;
                dataFileName = @"..\" + dataFileName;
            }

            return "";
        }
        #endregion

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support; do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.manualButton = new System.Windows.Forms.Button();
            this.autoButton = new System.Windows.Forms.Button();
            this.button1 = new System.Windows.Forms.Button();
            this.angleFormButton = new System.Windows.Forms.Button();
            this.button2 = new System.Windows.Forms.Button();
            this.button3 = new System.Windows.Forms.Button();
            this.SuspendLayout();
            // 
            // manualButton
            // 
            this.manualButton.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.manualButton.Location = new System.Drawing.Point(48, 24);
            this.manualButton.Name = "manualButton";
            this.manualButton.Size = new System.Drawing.Size(184, 23);
            this.manualButton.TabIndex = 0;
            this.manualButton.Text = "Manual Calculations";
            this.manualButton.Click += new System.EventHandler(this.manualButton_Click);
            // 
            // autoButton
            // 
            this.autoButton.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.autoButton.Location = new System.Drawing.Point(48, 64);
            this.autoButton.Name = "autoButton";
            this.autoButton.Size = new System.Drawing.Size(184, 23);
            this.autoButton.TabIndex = 1;
            this.autoButton.Text = "Auto Calculations";
            this.autoButton.Click += new System.EventHandler(this.autoButton_Click);
            // 
            // button1
            // 
            this.button1.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.button1.Location = new System.Drawing.Point(48, 104);
            this.button1.Name = "button1";
            this.button1.Size = new System.Drawing.Size(184, 23);
            this.button1.TabIndex = 2;
            this.button1.Text = "Controls as Keys";
            this.button1.Click += new System.EventHandler(this.button1_Click);
            // 
            // angleFormButton
            // 
            this.angleFormButton.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.angleFormButton.Location = new System.Drawing.Point(48, 144);
            this.angleFormButton.Name = "angleFormButton";
            this.angleFormButton.Size = new System.Drawing.Size(184, 23);
            this.angleFormButton.TabIndex = 3;
            this.angleFormButton.Text = "Angle Calculations";
            this.angleFormButton.Click += new System.EventHandler(this.angleFormButton_Click);
            // 
            // button2
            // 
            this.button2.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.button2.Location = new System.Drawing.Point(48, 184);
            this.button2.Name = "button2";
            this.button2.Size = new System.Drawing.Size(184, 23);
            this.button2.TabIndex = 4;
            this.button2.Text = "Auto Angle Calculations";
            this.button2.Click += new System.EventHandler(this.button2_Click);
            // 
            // button3
            // 
            this.button3.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.button3.Location = new System.Drawing.Point(48, 224);
            this.button3.Name = "button3";
            this.button3.Size = new System.Drawing.Size(184, 23);
            this.button3.TabIndex = 5;
            this.button3.Text = "Algebraic Expressions";
            this.button3.Click += new System.EventHandler(this.button3_Click);
            // 
            // SelectForm
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.BackColor = System.Drawing.Color.FromArgb(((int)(((byte)(237)))), ((int)(((byte)(240)))), ((int)(((byte)(246)))));
            this.ClientSize = new System.Drawing.Size(272, 270);
            this.Controls.Add(this.button3);
            this.Controls.Add(this.button2);
            this.Controls.Add(this.angleFormButton);
            this.Controls.Add(this.button1);
            this.Controls.Add(this.autoButton);
            this.Controls.Add(this.manualButton);
            this.Name = "SelectForm";
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen;
            this.Text = "Select Form Demo";
            this.Load += new System.EventHandler(this.SelectForm_Load);
            this.ResumeLayout(false);

		}
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
# if SyncfusionFramework1_1 || SyncfusionFramework2_0  
			Application.EnableVisualStyles();
# endif
			Application.Run(new SelectForm());
		}

		//used in automated testing
		Form[] forms = new Form[6];
		private void manualButton_Click(object sender, System.EventArgs e)
		{
			ManualCalcForm f = new ManualCalcForm();
			forms[0] = f;
			f.Show();
		}

		private void autoButton_Click(object sender, System.EventArgs e)
		{
			AutoCalcForm f = new AutoCalcForm();
			forms[1] = f;
			f.Show();
		}

		private void button1_Click(object sender, System.EventArgs e)
		{
			MoreComplexForm f = new MoreComplexForm();
			forms[2] = f;
			f.Show();
		}

		private void angleFormButton_Click(object sender, System.EventArgs e)
		{
			AngleForm f = new AngleForm();
			forms[3] = f;
			f.Show();
		}

		private void button2_Click(object sender, System.EventArgs e)
		{
			AutoAngleForm f = new AutoAngleForm();
			forms[4] = f;
			f.Show();
		}

		private void button3_Click(object sender, System.EventArgs e)
		{
			AlgebraicExpressions f = new AlgebraicExpressions();
			forms[5] = f;
			f.Show();
		}

		private void SelectForm_Load(object sender, System.EventArgs e)
		{
			if(Environment.CommandLine.EndsWith("automatedtest"))
			{
				this.manualButton.PerformClick();
				this.autoButton.PerformClick();
				this.button1.PerformClick();
				this.angleFormButton.PerformClick();
				this.button2.PerformClick();
				this.button3.PerformClick();
				Size sz = SystemInformation.PrimaryMonitorSize;;
				this.forms[0].Location = new Point(0,0);
				this.forms[0].Size = new Size(sz.Width/3, sz.Height/2);
				this.forms[1].Location = new Point(sz.Width/3+1,0);
				this.forms[1].Size = new Size(sz.Width/3, sz.Height/2);
				this.forms[2].Location = new Point(2*sz.Width/3+1,0);
				this.forms[2].Size = new Size(sz.Width/3, sz.Height/2);
				this.forms[3].Location = new Point(0,sz.Height/2+1);
				this.forms[3].Size = new Size(sz.Width/3, sz.Height/2);
				this.forms[4].Location = new Point(sz.Width/3+1,sz.Height/2+1);
				this.forms[4].Size = new Size(sz.Width/3, sz.Height/2);
				this.forms[5].Location = new Point(2*sz.Width/3+1,sz.Height/2+1);
				this.forms[5].Size = new Size(sz.Width/3, sz.Height/2);
				this.WindowState = FormWindowState.Minimized;
			}
		}

	}
}

File: Syncfusion_Sample/Windows/Calculate.Windows/Samples/2.0/Working With CalcQuick Demo/VB/SelectForm.vb
#Region "Copyright Syncfusion Inc. 2001 - 2006"
'
'  Copyright Syncfusion Inc. 2001 - 2006. All rights reserved.
'
'  Use of this code is subject to the terms of our license.
'  A copy of the current license can be obtained at any time by e-mailing
'  licensing@syncfusion.com. Any infringement will be prosecuted under
'  applicable laws. 
'
#End Region

Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms


''' <summary>
''' Summary description for SelectForm.
''' </summary>
Public Class SelectForm : Inherits System.Windows.Forms.Form
    Private WithEvents manualButton As System.Windows.Forms.Button
    Private WithEvents autoButton As System.Windows.Forms.Button
    Private WithEvents button1 As System.Windows.Forms.Button
    Private WithEvents angleFormButton As System.Windows.Forms.Button
    Private WithEvents button2 As System.Windows.Forms.Button
    Private WithEvents button3 As System.Windows.Forms.Button
    ''' <summary>
    ''' Required designer variable.
    ''' </summary>
    Private components As System.ComponentModel.Container = Nothing

    Public Sub New()
        '
        ' Required for Windows Form Designer support.
        '
        InitializeComponent()

        Try
            Dim ico As Icon = New Icon(FindFile("Icon", "sfgrid.ico"))
            Me.Icon = ico
        Catch
        End Try
        '
        ' TODO: Add any constructor code after InitializeComponent call
        '
    End Sub

#Region "FindFile Utility"
    ''' <summary>
    ''' Finds a file of the given name in the current directory or sibling "Data" directory.
    ''' If file is not found, the parent folder is checked until the file is found. This method
    ''' is used by our samples when they load data from a separate "Data" folder.
    ''' </summary>
    ''' <param name="dataDirName">The name of the "Data" folder.</param>
    ''' <param name="fileName">The filename to be searched.</param>
    ''' <returns>The full path of the file that was found; an empty string is returned if file is not found.</returns>

    Public Shared Function FindFile(ByVal dataDirName As String, ByVal fileName As String) As String
        dataDirName = dataDirName.TrimEnd("\"c, "/"c)
        ' Check both in parent folder and Parent\Data folders.
        Dim dataFileName As String = dataDirName & "\" & fileName
        Dim n As Integer
        For n = 0 To 9
            If System.IO.File.Exists(fileName) Then
                Return fileName
            End If
            If System.IO.File.Exists(dataFileName) Then
                Return dataFileName
            End If
            fileName = "..\" & fileName
            dataFileName = "..\" & dataFileName
        Next n

        Return ""
    End Function
#End Region

    ''' <summary>
    ''' Clean up any resources being used.
    ''' </summary>
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not components Is Nothing Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

#Region "Windows Form Designer generated code"
    ''' <summary>
    ''' Required method for Designer support; do not modify
    ''' the contents of this method with the code editor.
    ''' </summary>
    Private Sub InitializeComponent()
        Me.manualButton = New System.Windows.Forms.Button
        Me.autoButton = New System.Windows.Forms.Button
        Me.button1 = New System.Windows.Forms.Button
        Me.angleFormButton = New System.Windows.Forms.Button
        Me.button2 = New System.Windows.Forms.Button
        Me.button3 = New System.Windows.Forms.Button
        Me.SuspendLayout()
        '
        'manualButton
        '
        Me.manualButton.FlatStyle = System.Windows.Forms.FlatStyle.System
        Me.manualButton.Location = New System.Drawing.Point(48, 24)
        Me.manualButton.Name = "manualButton"
        Me.manualButton.Size = New System.Drawing.Size(184, 23)
        Me.manualButton.TabIndex = 0
        Me.manualButton.Text = "Manual Calculations"
        '
        'autoButton
        '
        Me.autoButton.FlatStyle = System.Windows.Forms.FlatStyle.System
        Me.autoButton.Location = New System.Drawing.Point(48, 64)
        Me.autoButton.Name = "autoButton"
        Me.autoButton.Size = New System.Drawing.Size(184, 23)
        Me.autoButton.TabIndex = 1
        Me.autoButton.Text = "Auto Calculations"
        '
        'button1
        '
        Me.button1.FlatStyle = System.Windows.Forms.FlatStyle.System
        Me.button1.Location = New System.Drawing.Point(48, 104)
        Me.button1.Name = "button1"
        Me.button1.Size = New System.Drawing.Size(184, 23)
        Me.button1.TabIndex = 2
        Me.button1.Text = "Controls as Keys"
        '
        'angleFormButton
        '
        Me.angleFormButton.FlatStyle = System.Windows.Forms.FlatStyle.System
        Me.angleFormButton.Location = New System.Drawing.Point(48, 144)
        Me.angleFormButton.Name = "angleFormButton"
        Me.angleFormButton.Size = New System.Drawing.Size(184, 23)
        Me.angleFormButton.TabIndex = 3
        Me.angleFormButton.Text = "Angle Calculations"
        '
        'button2
        '
        Me.button2.FlatStyle = System.Windows.Forms.FlatStyle.System
        Me.button2.Location = New System.Drawing.Point(48, 184)
        Me.button2.Name = "button2"
        Me.button2.Size = New System.Drawing.Size(184, 23)
        Me.button2.TabIndex = 4
        Me.button2.Text = "Auto Angle Calculations"
        '
        'button3
        '
        Me.button3.FlatStyle = System.Windows.Forms.FlatStyle.System
        Me.button3.Location = New System.Drawing.Point(48, 224)
        Me.button3.Name = "button3"
        Me.button3.Size = New System.Drawing.Size(184, 23)
        Me.button3.TabIndex = 5
        Me.button3.Text = "Algebraic Expressions"
        '
        'SelectForm
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.BackColor = System.Drawing.Color.FromArgb(CType(CType(237, Byte), Integer), CType(CType(240, Byte), Integer), CType(CType(246, Byte), Integer))
        Me.ClientSize = New System.Drawing.Size(272, 270)
        Me.Controls.Add(Me.button3)
        Me.Controls.Add(Me.button2)
        Me.Controls.Add(Me.angleFormButton)
        Me.Controls.Add(Me.button1)
        Me.Controls.Add(Me.autoButton)
        Me.Controls.Add(Me.manualButton)
        Me.Name = "SelectForm"
        Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen
        Me.Text = "Select Form Demo"
        Me.ResumeLayout(False)

    End Sub
#End Region

    ''' <summary>
    ''' The main entry point for the application.
    ''' </summary>
    <STAThread()> _
    Shared Sub Main()
#If ((SyncfusionFramework1_1 OrElse SyncfusionFramework2_0)) Then
Application.EnableVisualStyles
#End If
        Application.Run(New SelectForm())
    End Sub

    'used in automated testing
    Private forms As Form() = New Form(5) {}
    Private Sub manualButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles manualButton.Click
        Dim f As ManualCalcForm = New ManualCalcForm()
        forms(0) = f
        f.Show()
    End Sub

    Private Sub autoButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles autoButton.Click
        Dim f As AutoCalcForm = New AutoCalcForm()
        forms(1) = f
        f.Show()
    End Sub

    Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button1.Click
        Dim f As MoreComplexForm = New MoreComplexForm()
        forms(2) = f
        f.Show()
    End Sub

    Private Sub angleFormButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles angleFormButton.Click
        Dim f As AngleForm = New AngleForm()
        forms(3) = f
        f.Show()
    End Sub

    Private Sub button2_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button2.Click
        Dim f As AutoAngleForm = New AutoAngleForm()
        forms(4) = f
        f.Show()
    End Sub

    Private Sub button3_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button3.Click
        Dim f As AlgebraicExpressions = New AlgebraicExpressions()
        forms(5) = f
        f.Show()
    End Sub

    Private Sub SelectForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Environment.CommandLine.EndsWith("automatedtest") Then
            Me.manualButton.PerformClick()
            Me.autoButton.PerformClick()
            Me.button1.PerformClick()
            Me.angleFormButton.PerformClick()
            Me.button2.PerformClick()
            Me.button3.PerformClick()
            Dim sz As Size = SystemInformation.PrimaryMonitorSize

            Me.forms(0).Location = New Point(0, 0)
            Me.forms(0).Size = New Size(sz.Width / 3, sz.Height / 2)
            Me.forms(1).Location = New Point(sz.Width / 3 + 1, 0)
            Me.forms(1).Size = New Size(sz.Width / 3, sz.Height / 2)
            Me.forms(2).Location = New Point(2 * sz.Width / 3 + 1, 0)
            Me.forms(2).Size = New Size(sz.Width / 3, sz.Height / 2)
            Me.forms(3).Location = New Point(0, sz.Height / 2 + 1)
            Me.forms(3).Size = New Size(sz.Width / 3, sz.Height / 2)
            Me.forms(4).Location = New Point(sz.Width / 3 + 1, sz.Height / 2 + 1)
            Me.forms(4).Size = New Size(sz.Width / 3, sz.Height / 2)
            Me.forms(5).Location = New Point(2 * sz.Width / 3 + 1, sz.Height / 2 + 1)
            Me.forms(5).Size = New Size(sz.Width / 3, sz.Height / 2)
            Me.WindowState = FormWindowState.Minimized
        End If
    End Sub

End Class

